### PR TITLE
Add administration api and rely on it for instances cli

### PR DESCRIPTION
**Pull request by jinroh**
This PR adds an administration endpoint on a separate port (6060 by default).

This administration offers for now the possibility to handle instances through an HTTP API:

- `POST /instances/?Domain=...`
- `GET /instances/`
- `DELETE /instances/:domain`

The `instances` cli command uses has been changed to use this API.

### CODE REVIEW

File: cmd/instances.go [hunk 1]
```diff
@@ -27,6 +29,7 @@ For example, bob.cozycloud.cc is the instance of Bob. A single cozy-stack
 process can manage several instances.
 
 Each instance has a separate space for storing files and a prefix used to
+Each instance has a separate space for storing files and a prefix used to
```
Comment (nono) [inline cmd/instances.go]: Duplicate line

File: cmd/instances.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	"github.com/cozy/cozy-stack/instance"
+	"github.com/howeyc/gopass"
+	"github.com/spf13/cobra"
+)
+
+var flagLocale string
+var flagApps []string
+
+// instanceCmdGroup represents the instances command
+var instanceCmdGroup = &cobra.Command{
```
Comment (aenario) [inline cmd/instances.go]: [nitpick] we are already in the "instances" file, there is no need to put the word instance in all private variable names.

File: instance/instance.go [hunk 3]
```diff
@@ -68,6 +69,21 @@ func (i *Instance) Rev() string { return i.DocRev }
 // SetRev implements couchdb.Doc
 func (i *Instance) SetRev(v string) { i.DocRev = v }
 
+// SelfLink is used to generate a JSON-API link for the directory
```
Comment (aenario) [inline instance/instance.go]: directory -> instance

File: cmd/instances.go [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+		return nil, err
+	}
+	return doc.Data, nil
+}
+
+func init() {
+	instanceCmdGroup.AddCommand(addInstanceCmd)
+	instanceCmdGroup.AddCommand(lsInstanceCmd)
+	instanceCmdGroup.AddCommand(destroyInstanceCmd)
+	instanceCmdGroup.AddCommand(passwdInstanceCmd)
```
Comment (aenario) [inline cmd/instances.go]: passwd should not be a children of instanceCmdGroup

File: web/instances/instances.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package instances
```
Comment (aenario) [inline web/instances/instances.go]: may be we should move this file to a separate web/admin/instances hierarchy

File: cmd/instances.go [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+type instancesAPIData struct {
+	Data []*instanceData `json:"data"`
+}
+
+func instancesClient() *client {
+	var password []byte
+
+	if !config.IsDevRelease() {
+		var err error
+		fmt.Printf("Password:")
```
Comment (aenario) [inline cmd/instances.go]: If the user starting the cmd instances can read the AdminConfigFile, it is not necessary to ask the user to enter it manually.

File: cmd/instances.go (hunk 2 above)
Comment (jinroh) [inline cmd/instances.go]: but we are the the `cmd` package

File: cmd/instances.go (hunk 4 above)
Comment (jinroh) [inline cmd/instances.go]: yes I agree its a bit weird under the instances command... you think it should be a root command ?

File: cmd/instances.go (hunk 6 above)
Comment (jinroh) [inline cmd/instances.go]: I'm not sure I understand your comment. What config file are you talking about ? The file containing the hashed password ?

File: cmd/instances.go (hunk 2 above)
Comment (aenario) [inline cmd/instances.go]: ok, nevermind

File: cmd/instances.go (hunk 6 above)
Comment (aenario) [inline cmd/instances.go]: I did not pay attention to the fact the password was hashed. Nevermind.

**Comment by jinroh** [comment]
rebased and squashed

**Pull request merged**